Repo: Build break in local development on main

### Suggestion

🙃 

```plaintext
    ✖  nx run parser:build
$ tsc -b tsconfig.build.json
       src/parser.ts(108,5): error TS2322: Type '{ ecmaVersion: number | undefined; globalReturn: boolean | undefined; jsxPragma: string | null | undefined; jsxFragmentName: string | null | undefined; lib: Lib[] | undefined; sourceType: "script" | "module"; }' is not assignable to type 'AnalyzeOptions'.
         Object literal may only specify known properties, and 'ecmaVersion' does not exist in type 'AnalyzeOptions'.
```

```plaintext
josh ~/repos/typescript-eslint/packages/parser $ yarn build   
yarn run v1.22.19
$ tsc -b tsconfig.build.json
src/parser.ts:108:5 - error TS2322: Type '{ ecmaVersion: number | undefined; globalReturn: boolean | undefined; jsxPragma: string | null | undefined; jsxFragmentName: string | null | undefined; lib: Lib[] | undefined; sourceType: "script" | "module"; }' is not assignable to type 'AnalyzeOptions'.
  Object literal may only specify known properties, and 'ecmaVersion' does not exist in type 'AnalyzeOptions'.

108     ecmaVersion: options.ecmaVersion === 'latest' ? 1e8 : options.ecmaVersion,
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/packages/ast-spec/src/base/BaseNode.ts
+++ b/packages/ast-spec/src/base/BaseNode.ts
@@ -3,6 +3,8 @@ import type { AST_NODE_TYPES } from '../ast-node-types';
 import type { NodeOrTokenData } from './NodeOrTokenData';
 
 export interface BaseNode extends NodeOrTokenData {
+  type: AST_NODE_TYPES;
+
   /**
    * The parent node of the current node
    *
@@ -10,6 +12,4 @@ export interface BaseNode extends NodeOrTokenData {
    * while traversing.
    */
   // parent?: Node;
-
-  type: AST_NODE_TYPES;
 }

--- a/packages/types/package.json
+++ b/packages/types/package.json
@@ -33,7 +33,7 @@
     "build": "tsc -b tsconfig.build.json",
     "postbuild": "downlevel-dts dist _ts3.4/dist",
     "clean": "tsc -b tsconfig.build.json --clean",
-    "postclean": "rimraf dist && rimraf _ts3.4 && rimraf coverage",
+    "postclean": "rimraf dist && rimraf src/generated && rimraf _ts3.4 && rimraf coverage",
     "format": "prettier --write \"./**/*.{ts,mts,cts,tsx,js,mjs,cjs,jsx,json,md,css}\" --ignore-path ../../.prettierignore",
     "generate:lib": "../../node_modules/.bin/ts-node --files --transpile-only ../scope-manager/tools/generate-lib.ts",
     "lint": "nx lint",

--- a/tools/postinstall.ts
+++ b/tools/postinstall.ts
@@ -21,7 +21,10 @@ void (async function (): Promise<void> {
   // Install git hooks
   await $`yarn husky install`;
 
-  // // Build all the packages ready for use
+  // Clean any caches that may be invalid now
+  await $`yarn clean`;
+
+  // Build all the packages ready for use
   await $`yarn build`;
 })();
 

